framework/include/base/FEProblemBase.h

```diff
@@ -1064,6 +1064,9 @@ class FEProblemBase :
    */
   void reportMooseObjectDependency(MooseObject * a, MooseObject * b);
 
+  /// force the changed element lists to be cached after adaptivity steps
+  void requestCacheMeshChanged() { _cache_mesh_changed = true; }
```

Comment:
Small thing: but I don't really like this name... it doesn't mean anything.  Does it cache that the mesh has changed?  Does it request the cache for the mesh changed?  Does it turn on caching of when the mesh changed?

Oh - it causes the elements that have been changed to be cached!  😉 

How about: `forceCachingOfChangedElementsWhenMeshChanges()`... or something...